Fix deprecated system-pools option

vijay-vin opened the pull request:
"ibmcloud pi system-pools" command is deprecated now. Use "ibmcloud pi datacenter" command.
Also RHCOS image versioning has changed. Instead of mapping to OCP versions, now RHCOS images are mapped to RHEL version. So, I made this change too in the script.

vijay-vin committed:
Fix deprecated system-pools option

Signed-off-by: Vijayv <vijay.vinnakota@ibm.com>

prb112 commented on openshift-install-powervs:
```diff
@@ -61,6 +62,7 @@ EOF
 }
 
 RELEASE_VER=${RELEASE_VER:-"4.15"}
+RHCOS_RELEASE_VER=${RHCOS_RELEASE_VER:-"9.4"}
```
I'd lean towards setting 9.6.

taliandre49 replied:
I also agree, i believe 9.6 is consistent with some tutorial documents i have seen!

vijay-vin replied:
Updated the script to point to 9.6 version for RHCOS and 4.20 version for OCP.

prb112 commented on openshift-install-powervs:
```diff
@@ -61,6 +62,7 @@ EOF
 }
 
 RELEASE_VER=${RELEASE_VER:-"4.15"}
```
Might as well bump to 4.20

vijay-vin replied:
Updated the script to point to 9.6 version for RHCOS and 4.20 version for OCP.

clnperez commented:
@vijay-vin 
1.  I would put this in the Usage or Advanced Usage section where the variables are being discussed, e.g. https://github.com/openshift/installer/blob/release-4.19/data/data/coreos/rhcos.json. 
2. Yes that is confusing but it's in line with how RH is talking about these two things. You can add some verbiage in your update to make sure it's clear which heading goes with which var

vijay-vin committed:
resolving review comments

Signed-off-by: Vijayv <vijay.vinnakota@ibm.com>

vijay-vin commented:
I have updated the script to point to 9-6 version for RHCOS and 4.20 version for OCP. 

I have also updated the Readme file with the similar version changes, and also added the link that Christy mentioned in her review comments.

Prajyot-Parab commented on README.md:
```diff
@@ -117,7 +117,7 @@ Where <args>:
 Environment Variables:
   IBMCLOUD_API_KEY    IBM Cloud API key
-  RELEASE_VER         OpenShift release version (Default: 4.15)
+  RELEASE_VER         OpenShift release version (Default: 4.20)
```
Add `RHCOS_RELEASE_VER` to this list with appropriate information.

vijay-vin replied:
Updated the README file accordingly.

Prajyot-Parab commented on README.md:
```diff
@@ -190,12 +190,13 @@ Before running the script, you may choose to override some environment variables
+By default OpenShift version 4.20 is installed.
 
-If you want to install 4.14, then export the variable `RELEASE_VER`.
+If you want to install 4.19, then export the variable `RELEASE_VER`.
```
this wont work without exporting `RHCOS_RELEASE_VER` for 4.19 and older, even that needs to be exported properly. (add appropriate required instructions)

vijay-vin replied:
Updated the README file accordingly.

Prajyot-Parab commented on README.md:
```diff
@@ -214,8 +215,8 @@ You can avoid the interactive mode by having the required input files available
     rhel_image_name = "rhel-83"
     rhcos_image_name = "rhcos-415"
```
update this to 9-6 equivalent image to make it default with 4.20

vijay-vin replied:
Updated the README file accordingly.

vijay-vin committed:
resolving review comments of Yussufsh

Signed-off-by: Vijayv <vijay.vinnakota@ibm.com>

Prajyot-Parab reviewed (dismissed):
/lgtm
/hold for @yussufsh

prb112 commented on README.md:
```diff
@@ -211,11 +214,11 @@ You can avoid the interactive mode by having the required input files available
     service_instance_id = "123456abc-xzz-2223434343"
-    rhel_image_name = "rhel-83"
-    rhcos_image_name = "rhcos-415"
+    rhel_image_name = "RHEL9-SP4"
```
```suggestion
    rhel_image_name = "RHEL9-SP6"
```

Hey @vijay-vin maybe we should use SP6

vijay-vin replied:
Done

yussufsh commented on openshift-install-powervs:
```diff
@@ -1049,7 +1051,7 @@ function variables {
   [ -z "$ALL_OCP_VERSIONS" ] && error "No OCP versions found for version $RELEASE_VER... Ensure you have set correct RELEASE_VER"
 
-  ALL_SYSTEM_TYPES=$($CLI_PATH pi system-pools 2>/dev/null | grep "System Type" | awk '{print $3}' | sort | uniq)
+  ALL_SYSTEM_TYPES=$($CLI_PATH pi datacenter get ${ZONE} 2>/dev/null | grep "General" |cut -f2 -d\[ | cut -f1 -d\] | sort |sed 's/ /\n/g')
```
`sort` won't work here because the `pi datacenter get` return 1 liner o/p.
Maybe we can use the `--json` format?

prb112 replied:
@vijay-vin can you look at this?

vijay-vin replied:
sure. Let me look into it.

vijay-vin replied:
Fixed.

Survey Output before the fix:
[question] > Select the system type to use for cluster nodes: 
_1) s1022
2) e980
3) s922_

Survey output after the fix:
[question] > Select the system type to use for cluster nodes:
_1) e980
2) s922
3) s1022_

vijay-vin commented:
@prb112 and @yussufsh  I have now fixed the files based on your review comments

yussufsh reviewed (approved):
/lgtm
